AbcSmc.cpp

```diff
@@ -1110,13 +1110,10 @@ Row AbcSmc::sample_priors(const gsl_rng* RNG, Mat2D& posterior, int &posterior_r
 
 void AbcSmc::calculate_doubled_variances( int t ) {
     vector<RunningStat> stats(npar());
-
-    for (size_t i = 0; i < _predictive_prior[t].size(); i++) {
-        for (size_t j = 0; j < npar(); j++) {
-            int particle_idx = _predictive_prior[t][i];
-            double par_value = _particle_parameters[t](particle_idx, j);
-            stats[j].Push(par_value);
-        }
+    Mat2D pars = _particle_parameters[t](_predictive_prior[t], Eigen::placeholders::all);
```

Comment:
```suggestion
    Mat2D par_values = _particle_parameters[t](_predictive_prior[t], Eigen::placeholders::all);
```
pars --> par_values (here and elsewhere